### PR TITLE
Create new house endpoint

### PR DESCRIPTION
**Pull request by shadowas-py**

### CODE REVIEW

File: src/routes/createNewHouse.routes.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import createNewHouseController from '../controllers/createNewHouse.controllers';
```
Comment (FilipHalon) [inline src/routes/createNewHouse.routes.js:1]: To keep the naming convention/routes storing practices similar to the rest of the code prepared so far, you could consider renaming the file to `houses.routes.js` and store here all house related routes.

File: src/models/createNewHouseModel.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    area: {
+        type: Number,
+        require: true
+    },
+    price: {
+        type: Number,
+        required: true
+    },
+    meta: {
+        houseId: {
```
Comment (FilipHalon) [inline src/models/createNewHouseModel.js:70]: I think houses will need a not nested `_id` to make them more easily referable for other schemas.

File: src/models/createNewHouseModel.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            type: mongoose.Schema.Types.ObjectId,
+            default: new mongoose.Types.ObjectId
+        },
+        created: {
+            type: Date,
+            default: Date.now(),
+            immutable: true
+        },
+        modified: {
+            type: Date,
```
Comment (FilipHalon) [inline src/models/createNewHouseModel.js:80]: That could probably be also `default: Date.now(),`.

File: src/services/findAddress.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import createNewHouse from '../models/createNewHouseModel'
+
+export const findAddress = async (addressData) => {
```
Comment (FilipHalon) [inline src/services/findAddress.js:3]: It doesn't work exactly as expected. The error will be thrown if someone tries to save a new house with an address already in the database but if a not required part of the address is skipped next time, the house will be saved. Checking if a house hasn't been added would probably have to rely on a subset of address data consisting only required fields.

File: src/models/createNewHouseModel.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        floors:{
+            type: Number,
+            required: true
+        },
+        heating:{
+            type: String,
+            required: true
+        }
+    },
+    OtherFeatures:{
```
Comment (FilipHalon) [inline src/models/createNewHouseModel.js:54]: It would be more consistent if it was lowercased.

File: src/models/createNewHouseModel.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
+const mongoose = require('mongoose')
+
+const housesSchema = new mongoose.Schema({
+    owner: {
+        type: mongoose.Schema.Types.ObjectId,
+        required: true
+    },
+    address:{
```
Comment (FilipHalon) [inline src/models/createNewHouseModel.js:8]: The address data could also use a street suffix and street number fields. It's best not to keep the data corresponding to the fields not as a part of any of the existing fields as parsing them could be difficult.

**Review by FilipHalon — CHANGES_REQUESTED** [review]

File: src/routes/createNewHouse.routes.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+const createNewHouseRoute = (router) => {
+    router.post('/create-new-house', async (req, res) => {
+      const newHouse = await createNewHouseController(req.body)
+      try {
```
Comment (FilipHalon) [inline src/routes/createNewHouse.routes.js:6]: Unfortunately, this `try-catch` block doesn't make much difference as if there is an error in `createNewHouseController` the route will return a response with the status 201 and an error message in its body.

File: src/models/createNewHouseModel.js [hunk 8]
```diff
@@ -0,0 +1,103 @@
+const mongoose = require('mongoose')
+
+const housesSchema = new mongoose.Schema({
```
Comment (FilipHalon) [inline src/models/createNewHouseModel.js:3]: `houseSchema` will also need a status field which would store the information if a house was deleted, sold or is still active. You could assign the numbers 0, 1, 2 to the corresponding statuses and store them in the database.

Comment (FilipHalon) [inline src/models/createNewHouseModel.js:3]: You could consider naming the fields in the schema the same as they are called on the front-end to make saving and retrieving data as effortless as possible unless you think some fields on the front-end should change their names. You might possibly also reconsider embeddings in this schema to make them work best with the most common operations like reading basic house data to be displayed on the houses list and on the map.

File: src/routes/createNewHouse.routes.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+          res.status(201).json(newHouse)
+      } catch (err){
+          res.status(400).json({ message: err.message})
```
Comment (mgoniszewska) [inline src/routes/createNewHouse.routes.js:9]: The rest of us use special packages that are inserting statuses while using more readable names for it.
`const { StatusCodes }= require( 'http-status-codes');`

**Review by mgoniszewska — CHANGES_REQUESTED** [review]

File: src/services/findAddress.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    const {country, city, street, houseNr, flatNr} = {...addressData}
+    const address = await createNewHouse.findOne({
+      'address.country': country,
+      'address.city': city,
+      'address.street': street,
+      'address.houseNr': houseNr,
+      'address.flatNr': flatNr
+    })
+    if (address === null) return { status: 'success'};
```
Comment (FilipHalon) [inline src/services/findAddress.js:12]: Should it return anything? In the controller the result of this function isn't assigned to any variable.

**Review by FilipHalon — APPROVED** [review]
Please run `prettier` on all the files you changed.

**Pull request merged**